### PR TITLE
speedyspeech code adapt for npu

### PR DESCRIPTION
**Pull request by warrentdrew**
### PR types
New features

### PR changes
Others

### Describe
 SpeedySpeech code adaption for npu, add device support and training, synthesize, synthesize_e2e and inference scripts

### CODE REVIEW

File: paddlespeech/t2s/exps/synthesize.py [hunk 1]
```diff
@@ -219,12 +219,18 @@ def parse_args():
     )
     # other
     parser.add_argument(
-        "--ngpu", type=int, default=1, help="if ngpu == 0, use cpu or xpu.")
+        "--ngpu", type=int, default=1, help="if ngpu == 0, use cpu or xpu or npu.")
```
Comment (zxcd) [inline paddlespeech/t2s/exps/synthesize.py]: fix grammar, do not use too much or.

File: paddlespeech/t2s/exps/synthesize.py [hunk 2]
```diff
@@ -219,12 +219,18 @@ def parse_args():
...
     parser.add_argument(
         "--nxpu",
         type=int,
         default=0,
-        help="if wish to use xpu, set ngpu == 0 and nxpu > 0, and if ngpu == 0 and nxpu == 0, use cpu."
+        help="if wish to use xpu, set ngpu == 0 and nxpu > 0, and if ngpu == 0 and nxpu == 0 and nnpu == 0, use cpu."
```
Comment (zxcd) [inline paddlespeech/t2s/exps/synthesize.py]: help should be written more concisely

File: paddlespeech/t2s/exps/synthesize.py [hunk 3]
```diff
@@ -219,12 +219,18 @@ def parse_args():
...
+    )
+    parser.add_argument(
+        "--nnpu",
+        type=int,
+        default=0,
+        help="if wish to use npu, set ngpu == 0 and nxpu == 0 and nnpu > 0, and if ngpu == 0 and nxpu == 0 and nnpu == 0, use cpu."
```
Comment (zxcd) [inline paddlespeech/t2s/exps/synthesize.py]: same with above

File: paddlespeech/t2s/exps/syn_utils.py [hunk 4]
```diff
@@ -591,7 +591,8 @@ def get_predictor(
 
     config = inference.Config(
         str(Path(model_dir) / model_file), str(Path(model_dir) / params_file))
-    config.enable_memory_optim()
+    if device != "npu":
```
Comment (zxcd) [inline paddlespeech/t2s/exps/syn_utils.py]: need to check why can not support here.

File: paddlespeech/t2s/exps/synthesize.py (hunk 1 above)
Comment (warrentdrew) [reply paddlespeech/t2s/exps/synthesize.py]: done

File: paddlespeech/t2s/exps/synthesize.py (hunk 2 above)
Comment (warrentdrew) [reply paddlespeech/t2s/exps/synthesize.py]: done

File: paddlespeech/t2s/exps/synthesize.py (hunk 3 above)
Comment (warrentdrew) [reply paddlespeech/t2s/exps/synthesize.py]: done

File: paddlespeech/t2s/exps/syn_utils.py (hunk 4 above)
Comment (warrentdrew) [reply paddlespeech/t2s/exps/syn_utils.py]: done

File: paddlespeech/t2s/exps/synthesize.py [hunk 5]
```diff
@@ -245,10 +254,12 @@ def main():
...
     elif args.nxpu > 0:
         paddle.set_device("xpu")
-    elif args.ngpu == 0 and args.nxpu == 0:
+    elif args.nnpu > 0:
+        paddle.set_device("npu")
+    elif args.ngpu == 0 and args.nxpu == 0 and args.nnpu == 0:
         paddle.set_device("cpu")
     else:
-        print("ngpu or nxpu should >= 0 !")
+        print("ngpu, nxpu and nnpu should be >= 0")
```
Comment (zxcd) [inline paddlespeech/t2s/exps/synthesize.py:262]: or?

File: paddlespeech/t2s/exps/synthesize_e2e.py [hunk 6]
```diff
@@ -339,10 +348,12 @@ def main():
...
     elif args.nxpu > 0:
         paddle.set_device("xpu")
-    elif args.ngpu == 0 and args.nxpu == 0:
+    elif args.nnpu > 0:
+        paddle.set_device("npu")
+    elif args.ngpu == 0 and args.nxpu == 0 or args.nnpu == 0:
         paddle.set_device("cpu")
     else:
-        print("ngpu or nxpu should >= 0 !")
+        print("ngpu, nxpu and nnpu should be >= 0")
```
Comment (zxcd) [inline paddlespeech/t2s/exps/synthesize_e2e.py:356]: or?

**Review by zxcd — APPROVED** [review]
LGTM, pls fix it in next pr.

**Pull request merged**